completions/ipmitool

```diff
@@ -16,11 +16,14 @@ _ipmitool()
             return
             ;;
         -*d)
-            COMPREPLY=($(compgen -W "$(
-                command ls -d /dev/ipmi* /dev/ipmi/* /dev/ipmidev/* \
-                    2>/dev/null | command sed -ne 's/^[^0-9]*\([0-9]\{1,\}\)/\1/p'
-            )" \
-                -- "$cur"))
+            local -a files
+            _comp_expand_glob files '/dev/ipmi* /dev/ipmi/* /dev/ipmidev/*'
+            ((${#files[@]})) &&
+                COMPREPLY=($(compgen -W "$(
+                    printf '%s\n' "${files[@]}" |
+                        command sed -ne 's/^[^0-9]*\([0-9]\{1,\}\)/\1/p'
```

Comment:
Now that we have the candidates in a bash array, I suppose we could loop over and filter them in pure bash instead of doing this. Not a biggie, leaving up to you to decide what to do about it or to leave as is.

Reply:
Refactored in 24e095b9. We can actually implement it using `compgen -X` without a loop.